client/gutenberg/editor/utils.js

```diff
@@ -3,29 +3,37 @@
  * External dependencies
  */
 import apiFetch from '@wordpress/api-fetch';
-import { noop } from 'lodash';
-
-/**
- * Internal dependencies
- */
-import debugFactory from 'debug';
-
-const debug = debugFactory( 'calypso:gutenberg' );
+import wpcomProxyRequest from 'wpcom-proxy-request';
+import { noop, startsWith } from 'lodash';
 
 export const overrideAPIPaths = siteSlug => {
-	// TODO: no API support for now. We'll also need to handle authorization here.
-	apiFetch.use( () => noop );
-
 	const rootURL = 'https://public-api.wordpress.com/';
 	apiFetch.use( apiFetch.createRootURLMiddleware( rootURL ) );
 
-	// rewrite default API paths to match WP.com equivalents
-	// Example: /wp/v2/posts -> /wp/v2/sites/{siteSlug}/posts
 	apiFetch.use( ( options, next ) => {
-		const wpcomPath = `/wp/v2/sites/${ siteSlug }/` + options.path.replace( '/wp/v2/', '' );
-
-		debug( 'sending API request to: ', wpcomPath );
-
-		return next( { ...options, path: wpcomPath } );
+		if ( startsWith( options.path, '/wp/v2/sites/' ) ) {
+			// rewrite default API paths to match WP.com equivalents
+			// Example: /wp/v2/posts -> /wp/v2/sites/{siteSlug}/posts
+			const wpcomPath = `/wp/v2/sites/${ siteSlug }/` + options.path.replace( '/wp/v2/', '' );
```

Comment:
sorry for the nit - could't pass it by

```js
const wpcomPath = `/wp/v2/sites/${ siteSlub }/${ options.path.replace( '/wp/v2/', '' ) }`
```

odd seeing a mixture of template literals and string concatenation

Reply:
Good spot 😉 That's pretty easy to fix, once we got the bigger parts in. This experiment doesn't necessarily follow the best code practices, so there might be other things that can be improved or optimized.